src/modules/sensors/vehicle_accelerometer/VehicleAccelerometer.cpp

```diff
@@ -0,0 +1,225 @@
+	_sensor_selection_sub.register_callback();
+
+	return SensorCorrectionsUpdate(true);
+}
+
+void
+VehicleAccelerometer::Stop()
+{
+	Deinit();
+
+	// clear all registered callbacks
+	for (auto sub : _sensor_accel_sub) {
+		sub.unregister_callback();
+	}
+
+	_sensor_selection_sub.unregister_callback();
+}
+
+void
+VehicleAccelerometer::SensorBiasUpdate(bool force)
+{
+	if (_sensor_bias_sub.updated() || force) {
+		sensor_bias_s bias;
+
+		if (_sensor_bias_sub.copy(&bias)) {
+			// TODO: should be checking device ID
+			_bias = Vector3f{bias.accel_bias};
+		}
+	}
+}
+
+bool
+VehicleAccelerometer::SensorCorrectionsUpdate(bool force)
+{
+	// check if the selected accel has updated
+	if (_sensor_correction_sub.updated() || force) {
+
+		sensor_correction_s corrections{};
+		_sensor_correction_sub.copy(&corrections);
+
+		// TODO: should be checking device ID
+		if (_selected_accel == 0) {
+			_offset = Vector3f{corrections.accel_offset_0};
+			_scale = Vector3f{corrections.accel_scale_0};
+
+		} else if (_selected_accel == 1) {
+			_offset = Vector3f{corrections.accel_offset_1};
+			_scale = Vector3f{corrections.accel_scale_1};
+
+		} else if (_selected_accel == 2) {
+			_offset = Vector3f{corrections.accel_offset_2};
+			_scale = Vector3f{corrections.accel_scale_2};
+
+		} else {
+			_offset = Vector3f{0.0f, 0.0f, 0.0f};
+			_scale = Vector3f{1.0f, 1.0f, 1.0f};
+		}
+
+		// update the latest accel selection
+		if (_selected_accel != corrections.selected_accel_instance) {
+			if (corrections.selected_accel_instance < MAX_ACCEL_COUNT) {
+				// clear all registered callbacks
+				for (auto sub : _sensor_accel_sub) {
+					sub.unregister_callback();
+				}
+
+				const int accel_new = corrections.selected_accel_instance;
+
+				if (_sensor_accel_sub[accel_new].register_callback()) {
+					PX4_DEBUG("selected accel changed %d -> %d", _selected_accel, accel_new);
+					_selected_accel = accel_new;
+
+					return true;
+				}
+			}
+		}
+	}
+
+	return false;
+}
+
+void
+VehicleAccelerometer::ParametersUpdate(bool force)
+{
+	// Check if parameters have changed
+	if (_params_sub.updated() || force) {
+		// clear update
+		parameter_update_s param_update;
+		_params_sub.copy(&param_update);
+
+		updateParams();
+
+		// get transformation matrix from sensor/board to body frame
+		const matrix::Dcmf board_rotation = get_rot_matrix((enum Rotation)_param_sens_board_rot.get());
+
+		// fine tune the rotation
+		const Dcmf board_rotation_offset(Eulerf(
+				math::radians(_param_sens_board_x_off.get()),
+				math::radians(_param_sens_board_y_off.get()),
+				math::radians(_param_sens_board_z_off.get())));
+
+		_board_rotation = board_rotation_offset * board_rotation;
+	}
+}
+
+void
+VehicleAccelerometer::Run()
+{
+	perf_begin(_cycle_perf);
+	perf_count(_interval_perf);
+
+	// update corrections first to set _selected_accel
+	SensorCorrectionsUpdate();
+
+	sensor_accel_s sensor_accel;
+
+	if (_sensor_accel_sub[_selected_accel].update(&sensor_accel)) {
```

Comment:
`_selected_accel` *may* still be -1 if the sensor_accel callback registration failed

Reply:
I'll make it an unsigned integer (initialized to zero), and force the first update.